### ISSUE TITLE
Update Side Nav Mobile Trigger to Respect Semantic Page Structure

### ISSUE THREAD

**Issue by dv297** [department-of-veterans-affairs/va.gov-team#131847]
### Description

The Side Navigation component utilizes the VADS Side Navigation component, which under the hood utilizes the USWDS Accordion component. 

Based on the current implementation, the trigger for expanding the Side Nav is a button that is nested underneath an "h2" element.

<img width="1059" height="54" alt="Image" src="https://github.com/user-attachments/assets/b5435023-dfae-41f1-9d45-e1a138b2e9df" />

This h2 element disrupts the semantics of the page, as the page looks like this from top to bottom

```html
<body>
  <h2>
    <button>Open Side Nav</button>
  </h2>
  <h1>File for disability compensation</h1>
  <h3>Progress Bar</h3>
</body>
```

This impacts screen readers' ability to interpret the page for accessibility purposes.

### How severe is the problem?

- [ ] Critical - A Veteran is blocked or significantly hindered from submitting the form, data is lost, or claim and evidence test coverage is missing. Fix ASAP.
- [x] Moderate - A Veteran is not blocked, but the happy path is interrupted. Fix after critical bugs.
- [ ] Minor - Cosmetic bugs and minor language inconsistencies. Defer to sprint planning.

### Desired behavior

If possible, the accordion trigger should not utilize a heading.  Wrapping the entire Side Nav in the `<aside />` tag could also denote that the navigation in the side nav is separate but tangentially related to the main content.

https://html.spec.whatwg.org/multipage/sections.html#the-aside-element

### Tasks

- Update the Side Nav button so that it does not add an additional h2 to the page.
- Wrap the Side Nav with an aside.

### Acceptance Criteria

- [ ] Desired behavior is met
- [ ] Desired behavior is validated using screen reader
- [ ] Code reviewed and approved
- [ ] Bugfix tested and reviewed by Design

### Definition of Done

- [ ] Meets acceptance criteria
- [ ] Reviewed and approved by product and/or design

### Engineering

- [ ] All tests pass
- [ ] New functionality is covered by unit tests
- [ ] Logging and monitoring are implemented (if applicable)

### Code Review & Pull Requests

- [ ] PR includes Local testing steps
- [ ] PR includes Flipper/testing state details (if applicable)
- [ ] PR includes Author's local proof of submission screenshot
- [ ] Copilot review completed and feedback addressed
- [ ] Internal reviewer approved
- [ ] Internal reviewer added local proof of submission screenshot
- [ ] Code functionality verified on Staging after merge

### Refinement Checklist!

- [ ] Added description, tasks, and acceptance criteria
- [ ] Added estimate
- [ ] Labeled with Practice Area (engineer, design, product, data science)
- [ ] Labeled with issue type and characteristics of the ticket (bug, accessibility, request, discovery, documentation, research, content, ux testing, front-end, back-end, datadog, etc.)
- [ ] Added any other relevant project fields (team, OCTO priority...)
- [ ] Added an Epic or Super Epic
